SPEC.md

```diff
@@ -0,0 +1,150 @@
+
+Until the next section, all keys will be automatically prepended the section name and a `.`, or with nothing (if the section name is the empty string).
+
+### Key-(Raw)Value Pairs
+A key-value and key-rawvalue pair look very similar.
+It's a key, an equals, and a value, with arbitrary amounts of spaces before and after the `=`.
+
+A value may contain any UTF-8 characters except vertical whitespace or the `#` sign (note: this is expanded upon in ini-compatibility).
+It may not start or end with whitespace, or start with a backtick.
+
+A rawvalue starts with a backtick and ends with a backtick, but may contain any UTF-8 characters besides those.
+If a backtick needs to be inserted into a raw value, a backtick may be escaped with a second one.
+
+Theoretically, this means that both values and raw values may contain arbitrary bytes, but the specification is exclusively intended to be used with UTF-8 data.
+Implementations are not required to validate it, though.
+
+### Comments
+A comment starts with a `#` (note: this is expanded upon in ini-compatibility) and ends at the next vertical whitespace.
+A comment may start on its own line, at any point "inside" of a value (thus ending it), after a section declaration, or after a raw value.
+It is impossible to have comments inside of a raw value, or inside of a section declaration.
+
+## Ini-Compatibility
+For compatibility with INI, the `;` character may also be used for comments, i.e treated as if it were a `#`.
+
+## Extensions
+All extensions are fully optional:
+* They are not part of "CNI", but are to be considered common extensions library authors may choose to add into their libraries, or application authors choose to make use of.
+* Library authors may implement any amount of extensions they wish.
+* Application authors should only mention any extensions that are important to the functioning of their application.
+* Application end-users should not rely on any extensions being present, unless the application they use makes it clear that said extension is available.
+ 
+### Flexspace
+In implementations with the "flexspace" extension, whitespace within statements is not significant.
+This means that you may have arbitrary whitespace (not only spaces, but also tabs, newlines, zero-width spaces, and so on) in the following locations:
+* Between the opening section bracket, the section name, and the closing section bracket.
+* Between the key in a key-(raw)value pair, the `=`, and the (raw)value.
+* Between a (raw) value and a comment.
+
+### Tabulation
+In implementations with the "tabulation" extension, whitespace before and after statements is not significant.
+This means you may have arbitrary horizontal whitespace in the following locations:
+* In front of a section declaration.
+* In front of a key-(raw)value pair.
+* In front of a comment.
+* Between a key-rawvalue pair and the next statement.
+* Between a section declaration and the next statement.
```

Comment:
Couldn't this be simplified to "at the start of any line"?

Reply:
The last two translate to something beyond that. The first 3 could, though, I'll think about it.